src/state/image-dataset/json-dataset/index.ts

```diff
@@ -92,31 +89,111 @@ class JsonRequest implements ImageDataset {
 
         // for now this particular file must be kept up to date.
         // dev-aics-dtp-001/cfedata is the designated repository of internal cfe data sets
+        // this datasets.json contains a list of the names of subdirectories (strings),
+        // each of which is expected to contain a dataset.json file
         this.listOfDatasetsDoc =
-            "http://dev-aics-dtp-001.corp.alleninstitute.org/cfedata/datasets.json";
+            "http://dev-aics-dtp-001.corp.alleninstitute.org/cfedata/cell-feature-data/datasets.json";
     }
 
     public getAvailableDatasets = () => {
-        return axios
-            .get(`${this.listOfDatasetsDoc}`)
-            .then((metadata: AxiosResponse) => metadata.data);
+        //const megasets: Megaset[] = [];
+        return axios.get(`${this.listOfDatasetsDoc}`).then((metadata: AxiosResponse) => {
+            const datadir = this.listOfDatasetsDoc.substring(
+                0,
+                this.listOfDatasetsDoc.lastIndexOf("/")
+            );
+
+            return Promise.all(
+                (metadata.data as string[]).map((datasetdir) => {
+                    const datasetdirpath = `${datadir}/${datasetdir}`;
+                    const fullurl = `${datasetdirpath}/dataset.json`;
+                    return axios.get(fullurl).then((dataset: AxiosResponse) => {
+                        const megaset = dataset.data as Megaset;
+                        if (megaset.datasets) {
+                            ///////////////////////////
+                            // TODO handle megasets from json flat file data
+                            ///////////////////////////
+                            console.log("has datasets; unexpected; dropping");
+                            return {} as Megaset;
+                        } else {
+                            const topLevelJson = { ...megaset } as unknown as DatasetMetaData;
+                            const megasetInfo = { ...megaset };
+                            megasetInfo.title = topLevelJson.title;
+                            megasetInfo.publications = megaset.publications || [];
+                            megasetInfo.extra = megaset.extra || "";
+                            const id = `${topLevelJson.name}_v${topLevelJson.version}`;
+                            megasetInfo.name = id;
```

Comment:
This block starts with three copies of the same object (`megaset`, `megasetInfo`, `topLevelJson`) and copies properties between them.

1. I see `megasetInfo` is eventually returned from here, and it looks like `topLevelJson` is there to keep an unmodified copy around to copy into `megasetInfo.datasets` below. Could `megaset` be copied in directly?
2. Looks like the title assignment on 121 may be redundant.
3. Suggestion: some of the remaining property assignments which just apply defaults could happen at assignment of `megasetInfo`, e.g.
```javascript
const megasetInfo = {
    publications: megaset.publications || [],
    extra: megasetInfo.extra || "",
    // ...
    ...megaset
};
```

Reply:
I believe this is now addressed. It is much simpler now.